File: 2019/01.php
#!/usr/bin/php
<?php

// Advent of Code 2019. Day One. http://adventofcode.com/2019/day/1

echo 'Advent of Code 2019, Day One.' . PHP_EOL;

$time_start = microtime(true);

$out1 = $out2 = 0;

$instructions = file('01.txt', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES) or die('No file.' . PHP_EOL);

// Part One.
foreach ($instructions as $line) {
    $out1 += (int) ($line / 3) - 2;
}

// Part Two.
foreach ($instructions as $line) {
    while (true) {
        $line = (int) ($line / 3) - 2;
        if ($line <= 0) {
            break;
        }
        $out2 += $line;
    }
}

// Part One: 3363929
// Part Two: 5043026
echo "Part One:\t" . $out1 . PHP_EOL;
if ($out2) {
    echo "Part Two:\t" . $out2 . PHP_EOL;
}
echo "Memory usage:\t" . number_format(memory_get_peak_usage() / 1024) . 'Kb' . PHP_EOL;
// echo "Time taken:\t" . number_format(microtime(true) - $time_start, 6) . ' seconds' . PHP_EOL;
echo "Time taken:\t" . number_format((microtime(true) - $time_start) * 1000, 3) . ' ms' . PHP_EOL;

File: 2019/01.py
#!/usr/bin/python3

# Advent of Code 2019. Day One. http://adventofcode.com/2019/day/1

import sys, time

print ("Advent of Code 2019, Day One.")

timestart = time.time()

out1 = out2 = 0

with open('01.txt') as instructions_file:
    # instructions = instructions_file.readlines()
    instructions = instructions_file.read().splitlines()

# Part One.
for line in instructions:
    # print(line.rstrip())
    # out1 = out1 + int( int(line.rstrip()) / 3 - 2 )
    out1 += int( int(line) / 3 - 2 )

# Part Two.
for line in instructions:
    while True:
        line = int( int(line) / 3 - 2 )
        if line <= 0:
            break
        out2 += line

timetaken = time.time() - timestart

# Part One: 3363929
# Part Two: 5043026
print("Part One:\t", out1)
if out2:
    print("Part Two:\t", out2)
print("Time taken:\t", round(timetaken * 1000, 3), 'ms')
